consumer/template_group_writer.go

```diff
@@ -0,0 +1,164 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"text/template"
+
+	"github.com/goat-project/goat/consumer/wrapper"
+)
+
+const (
+	templateFileName = "vm.tmpl"
+	templateName     = "VMS"
+	filenameFormat   = "%014d"
+)
+
+type vmsTemplateData struct {
+	Vms []interface{}
+}
+
+// TemplateGroupWriter converts each record to template and writes it to file.
+// Multiple records may be written into a single file.
+type TemplateGroupWriter struct {
+	outputDir    string
+	templatesDir string
+	countPerFile uint64
+	template     *template.Template
+	records      []interface{}
+}
+
+// NewTemplateGroupWriter creates a new TemplateGroupWriter.
+func NewTemplateGroupWriter(outputDir, templatesDir string, countPerFile uint64) TemplateGroupWriter {
+	return TemplateGroupWriter{
+		outputDir:    outputDir,
+		templatesDir: templatesDir,
+		countPerFile: countPerFile,
+		template:     nil,
+		records:      make([]interface{}, countPerFile),
+	}
+}
+
+func ensureDirectoryExists(path string) error {
+	err := os.MkdirAll(path, os.ModeDir)
```

Comment:
I think that `os.ModeDir` does not allow to read from or write to the directory.